### ISSUE TITLE
report_winner function doesn't work inside a group stage

### ISSUE THREAD

**Issue by 3rdCore** [fp12/achallonge#10]
`report_winner` function seems not to work correctly while being used for matches inside a group stage. The associated `group_player_ids` for a player inside a group is not understanted by the function `report_winner(..)`. This function expects a player with the `id` attribute equals to `group_player_ids` but get a player with his original id (outsite group stage) and crash. 

example, with 2 objects `player` and `match` : 
```
In [68]: player
Out[68]: <challonge.participant.Participant at 0x23866bb9c40>

In [69]: player.id
Out[69]: 152547829
In [70]: player.group_player_ids
Out[70]: [17666281] 

In [72]: match.player1_id
Out[72]: 17666281

await match.report_winner(player,score)

Error call : `APIException: ({'errors': ['Winner ID must be one of the player/participant IDs']}, 422, 'Unprocessable Entity', 'tournaments/10145851/matches/247328813', [('match[scores_csv]', '1-0'), ('match[winner_id]', '152547845')])
```

**Comment by fp12** [comment]
Hello,

I think you need to reverse the arguments and pass the id directly, not the player itself.
Could you try:
`await match.report_winner(score, player.id)`

Please let me know!

**Comment by fp12** [comment]
Thanks for reporting the issue!
I need to test more for group stages it seems...
I'll look into it!

**Comment by fp12** [comment]
@3rdCore did you create the group stage via API/achallonge?
If so, could you share the code? Thanks!

**Comment by 3rdCore** [comment]
I created the tournament using the web interface and I then call the tournament using its token.
Every thing was then done using the Challonge API. This is my main function : 
```python

  my_user = await challonge.get_user(my_username, my_api_key)
  my_tournament_id = 'ycpl6a79'
  my_tournament = await my_user.get_tournament(my_tournament_id)
  print(my_tournament.name, my_tournament.full_challonge_url)
  #if not (my_tournament.state =='pending'): 
  #    print("tournament reseted")
  #    await my_tournament.reset()
  #print("Beginning of the tournament !")
  await my_tournament.start()
  matches = await my_tournament.get_matches()
  players = await my_tournament.get_participants()
  player_index = {players[i].id:i for i in range(0,len(players))}

  for match in matches:
          player1 = await my_tournament.get_participant(match.player1_id)
          player2 = await my_tournament.get_participant(match.player2_id)
          #print("players id : ", player1.id, " ", player2.id)
          #player1.id = player1.  [0]
          #player2.id = player2.group_player_ids[0]
          #print("players id : ", player1, " ", player2.id)
          winner, score = get_winner_and_score(player1, player2)
          await match.report_winner(winner,score)
```

**Comment by fp12** [comment]
ok, so with `my_tournament_id = 'ycpl6a79'`, I'm assuming you set up the (group) tournament on the website and you are then updating the results through the API, right?

**Comment by 3rdCore** [comment]
That's right. First of all, the function `start()` doesn't start the group phase so we have to manually start the group phase on the website ( with the button ) otherwise the tournament is stuck in an in-between phase where the user can't modify anything. Should I try to totally setup a tournament through the API ? Have you understand the difference between .id an. .group_player_id ? Thanks for your help. On Aug 17, 2021 04:53, Fabien Poupineau ***@***.***> wrote:
ok, so with my_tournament_id = 'ycpl6a79', I'm assuming you set up the (group) tournament on the website and you are then updating the results through the API, right?

**Comment by 3rdCore** [comment]
That's right. I set up the tournament ( number of player, tournament tree.. ) online, does it come from this ? 

First of all, the function `start()` doesn't start the group phase so we have to manually start the group phase on the website ( with the button ) otherwise the tournament is stuck in an in-between phase where the user can't modify anything. 

Should I try to totally setup a tournament through the API ? 

Have you understand the difference between `.id` an. `.group_player_id` ? 

Thanks for your help.

**Comment by 3rdCore** [comment]
do you guys figured out what could be the problem ?

**Comment by fp12** [comment]
No, sorry, I haven't looked into it yet.
And since we can't do groups from the API, it makes testing a bit harder/longer...

**Comment by 3rdCore** [comment]
I've managed to update the scores using the function `report_tie()`. This function makes it possible to give un-tie scores and therefore undirectly declare a winner (it would appear as a tie on the web interface, but will still rank players in order of their score.)

This is obviously not a solution but could help someone else stuck in the same situation as me.
